docs/developer/payments.mdx

```diff
@@ -0,0 +1,578 @@
+---
+title: Payments
+---
+
+## Introduction
+
+The below process describes the key milestones in the payment process flow in Saleor.
+Additional steps may also occur along the way; however, the purpose of this instruction is to
+deliver a base reference for the user to work with.
+
+Saleor distinguishes two different ways of processing a payment:
+
+- By using a [plugin](extending/overview#plugins) embedded in the Saleor code.
+- By using a [Saleor App](extending/overview#apps-recommended) (recommended).
+
+## Saleor App (recommended)
+
+The flow where the payment process is handled by an external Saleor App such as [saleor-checkout](https://github.com/saleor/saleor-checkout).
+Saleor expects to receive a result of the payment transaction processed by Saleor App.
+
+### Creating transaction
+
+:::info
+
+This feature was introduced in **Saleor 3.4**.
+
+:::
+
+:::caution
+
+This feature is currently in **Feature Preview**. This part of Saleor is not complete
+and subject to change but is available to experiment and provide feedback.
+
+:::
+
+Transaction stores details of a payment transaction attached to order or checkout:
+
+The [transactionCreate](../api-reference/mutations/transaction-create) mutation takes the following arguments:
+
+- `id`: The ID of the checkout or order.
+- `transaction`: Input data required to create a new transaction object.
+- `transactionEvent`: Data that defines a transaction event. Can be used to provide more context about the current state of the transaction.
+
+The following example shows how the [transactionCreate](../api-reference/mutations/transaction-create) mutation
+is used to create a new transaction.
+The transaction was authorized, and payment was made with a credit card. The actions that can be called from Saleor are: `VOID` and `CHARGE`.
+The authorized amount is $99.
+
+#### Saleor 3.12+
+
+```graphql
+mutation {
+  transactionCreate(
+    id: "Q2hlY2tvdXQ6MWQzNmU5YzctYWEwYS00NzM5LTk0MGQtNzdjNmU4Mjc5YmQ0"
+    transaction: {
+      name: "Credit card"
+      message: "Authorized"
+      pspReference: "PSP-ref123"
+      availableActions: [VOID, CHARGE]
```

Comment:
I think that since the VOID will be deleted, we should use CANCEL in the examples